### PR TITLE
feat: dynamic errors spans payload & span prioritization

### PR DESCRIPTION
**Pull request by nadav3396**

### CODE REVIEW

File: src/reporter.ts [hunk 1]
```diff
@@ -115,22 +114,61 @@ export function scrubSpans(resultSpans: any[]) {
...
+  spans.sort(spansPrioritySorter);
+  let currentSize = 0;
+  const spansToSendSizes = {};
+  const spansToSend = {};
+
+  // First we try to take only the spans metadata
+  for (let index = 0; index < spans.length; index++) {
+    const spanMetadata = getSpanMetadata(spans[index]);
+    spansToSendSizes[index] = 0;
+    if (spanMetadata == undefined) continue;
```
Comment (harelmo-lumigo) [inline src/reporter.ts:128]: Not much of a difference in this case, but `===` is the de-facto standard:
```suggestion
    if (spanMetadata === undefined) continue;
```

File: src/spans/awsSpan.ts [hunk 2]
```diff
@@ -66,6 +68,79 @@ export const getSpanInfo = (): SpanInfo => {
...
+  const span1Priority = getSpanPriority(span1);
+  const span2Priority = getSpanPriority(span2);
+
+  if (span1Priority < span2Priority) {
+    return -1;
+  } else if (span1Priority > span2Priority) {
+    return 1;
+  } else {
+    return 0;
+  }
```
Comment (harelmo-lumigo) [inline src/spans/awsSpan.ts:94]:
```suggestion
  return Math.sign(span1Priority - span2Priority)
```

File: src/spans/awsSpan.ts [hunk 3]
```diff
@@ -66,6 +68,79 @@ export const getSpanInfo = (): SpanInfo => {
...
+};
+
+export const getSpanMetadata = (span: any): any => {
+  const spanCopy = JSON.parse(JSON.stringify(span));
```
Comment (harelmo-lumigo) [inline src/spans/awsSpan.ts:98]: So the benchmark is cool with that? :)

**Review by harelmo-lumigo — APPROVED** [review]

File: src/spans/awsSpan.ts (hunk 3 above)
Comment (nadav3396) [reply src/spans/awsSpan.ts:92]: yes, So I think we are good. (We do this logic only in rare cases so most of the time it will not run at all

**Pull request merged**